compile.c

```diff
@@ -1986,6 +1987,16 @@ cdhash_cmp(VALUE val, VALUE lit)
     else if (tlit == T_FLOAT) {
         return rb_float_cmp(lit, val);
     }
+    else if (tlit == T_RATIONAL) {
+        const struct RRational *rat1 = RRATIONAL(val);
+        const struct RRational *rat2 = RRATIONAL(lit);
+        return cdhash_cmp(rat1->num, rat2->num) && cdhash_cmp(rat1->den, rat2->den);
```

Comment:
Shouldn't this be "||" since the cmp functions return 0 when equivalent?

Reply:
@Math2 good catch! fixed in https://github.com/ruby/ruby/pull/4469/commits/b6c6e58b8b46d72899e5b4f01052de1f9f9b0dd5